recipes-app/node-red-gpio/files/0008-mraa-gpio-Add-support-for-IOT2050-SM.patch

```diff
@@ -0,0 +1,352 @@
++                $('#btype').text(RED.settings.mraaGpioAinBoardType);
++                $('#ver-tip').text(RED.settings.mraaGpioAinMraaVersion);
++            }
++        });
++    }
+ </script>
+ 
+ <script type="text/html" data-template-name="mraa-gpio-ain">
+diff --git a/hardware/intel/mraa-gpio-din.html b/hardware/intel/mraa-gpio-din.html
+index 444bfa9c67eb..4906124f3bba 100644
+--- a/hardware/intel/mraa-gpio-din.html
++++ b/hardware/intel/mraa-gpio-din.html
+@@ -21,13 +21,23 @@
+             return this.name?"node_label_italic":"";
+         },
+         oneditprepare: function() {
+-            if (RED.settings.mraaGpioDinBoardType === "IOT2050") {
+-                for (i of [...Array(20).keys()].slice(14)) {
+-                    // from D14 to D19
++            if (RED.settings.mraaGpioDinBoardType === "IOT2050 SM") {
++                $('#node-input-pin').append($("<option></option>").attr("value", 0).text("USER button"));
++            } else {
++                for (i of [...Array(14).keys()]) {
++                    // from D0 to D13
+                     $('#node-input-pin').append($("<option></option>").attr("value", i).text("D" + i));
+                 }
+-                $('#node-input-pin').append($("<option></option>").attr("value", 20).text("USER button"));
++
++                if (RED.settings.mraaGpioDinBoardType === "IOT2050") {
++                    for (i of [...Array(20).keys()].slice(14)) {
++                        // from D14 to D19
++                        $('#node-input-pin').append($("<option></option>").attr("value", i).text("D" + i));
++                    }
++                    $('#node-input-pin').append($("<option></option>").attr("value", 20).text("USER button"));
++                }
+             }
++
+             $('#node-input-pin').val(this.pin);
+             $('#type-tip').text(RED.settings.mraaGpioDinBoardType);
+             $('#ver-tip').text(RED.settings.mraaGpioDinMraaVersion);
+@@ -40,20 +50,6 @@
+         <label for="node-input-pin"><i class="fa fa-circle"></i> Pin</label>
+         <select type="text" id="node-input-pin" style="width: 250px;">
+             <option value='' disabled selected style='display:none;'><span data-i18n="rpi-gpio.label.selectpin"></span></option>
+-            <option value="0">D0</option>
+-            <option value="1">D1</option>
+-            <option value="2">D2</option>
+-            <option value="3">D3</option>
+-            <option value="4">D4</option>
+-            <option value="5">D5</option>
+-            <option value="6">D6</option>
+-            <option value="7">D7</option>
+-            <option value="8">D8</option>
+-            <option value="9">D9</option>
+-            <option value="10">D10</option>
+-            <option value="11">D11</option>
+-            <option value="12">D12</option>
+-            <option value="13">D13</option>
+         </select>
+     </div>
+     <div class="form-row">
```

Comment:
So the html file is still evaluated by Node RED, even if the js file does not register the node?

Reply:
Yep, that's right, and that's why I have to add the model condition to the html file.